daemon/logger/jsonfilelog/read.go

```diff
@@ -113,14 +113,16 @@ func (l *JSONFileLogger) readLogs(logWatcher *logger.LogWatcher, config logger.R
 }
 
 func tailFile(f io.ReadSeeker, logWatcher *logger.LogWatcher, tail int, since time.Time) {
-	var rdr io.Reader = f
+	var rdr io.Reader
 	if tail > 0 {
 		ls, err := tailfile.TailFile(f, tail)
 		if err != nil {
 			logWatcher.Err <- err
 			return
 		}
 		rdr = bytes.NewBuffer(bytes.Join(ls, []byte("\n")))
+	} else {
+		rdr = f.(io.Reader)
```

Comment:
I'll let this build run to see if everything is happy and then make that change.